ebos/eclgenericcpgridvanguard.cc

```diff
@@ -291,6 +298,19 @@ void EclGenericCpGridVanguard<ElementMapper,GridView,Scalar>::doCreateGrids_(Ecl
         auto &field_props = eclState.fieldProps();
         const_cast<FieldPropsManager&>(field_props).reset_actnum(actnum);
     }
+
+#if HAVE_MPI
+    {
+        EclMpiSerializer ser(Dune::MPIHelper::getCollectiveCommunication());
+        ser.broadcast(removed_cells);
```

Comment:
Oooh, this serializer concept is spreading everywhere. I always assumed that this would only be used for EclipseState.

Wouldn't it be easier to just do `grid_.comm().broadcast(removed_cells.data(), removed_cells.size(), 0)`? Then you could also kill the pack/unpack code.

Additionally the PR removing MPI_COMM_WORLD would work more nicely with this.

Reply:
> Wouldn't it be easier to just do `grid_.comm().broadcast(removed_cells.data(), removed_cells.size(), 0)`? Then you could also kill the pack/unpack code.

> Additionally the PR removing MPI_COMM_WORLD would work more nicely with this.

I strongly support this. Any new use of MPI_COMM_WORLD (directly or through using a default collective communication object) must be replaced anyway.

Reply:
> Wouldn't it be easier to just do
> ```C++
> grid_.comm().broadcast(removed_cells.data(), removed_cells.size(), 0)
> ```
> ?

Maybe.  Do I need to broadcast the size in a separate call first or does that happen implicitly.

> Then you could also kill the pack/unpack code.

That's an obvious benefit.

On a separate note, why do we have three different frameworks for communicating between ranks?

Reply:
I've pushed an update that switches to using the `comm().broadcast()` method.